CleanCode.SolidPrinciples/SingleResponsibility/StringHolidayCountReferences.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CleanCode.SolidPrinciples
+{
+    class StringHolidayCountReferences
+    {
+        //------------------------------------ MEMBERS ---------------------------------------//
+        private string[] references;
+
+        //------------------------------------ METHODS ---------------------------------------//
+                                            //constructors
+        StringHolidayCountReferences()
+        {
+            references = null;
+        }
+        StringHolidayCountReferences(string reference)
+        {
+            //in case of 'shared/static' property
+            for(int a=0; a<references.GetLength(0); a++)
+            {
+                if (references[a] == null)
+                {
+                    references[a] = reference;
+                    break;
+                }
+                    
+            }
+        }
+        StringHolidayCountReferences(string[] references)
+        {
+            for (int a = 0; a < references.GetLength(0); a++)
+                this.references[a] = references[a];
```

Comment:
possible overflow of array, what if referances (taken as argument) will have much more elements then this.references?